onnx/defs/nn/defs.cc

```diff
@@ -1229,4 +1225,91 @@ ONNX_OPERATOR_SET_SCHEMA(
         .SetDoc(LRN_ver1_doc)
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
-} // namespace ONNX_NAMESPACE
+ONNX_OPERATOR_SET_SCHEMA(
+    Conv_Integer,
+    8,
+    OpSchema()
+        .SetDoc(
+            "The integer convolution operator consumes an input tensor, a filter and a padding value, and computes the output. "
+            "The production MUST never overflow. The accumulation may overflow if and only if in 32 bits.")
+        .Input(
+            0,
+            "X",
+            "Input data tensor from previous layer; "
+            "has size (N x C x H x W), where N is the batch size, "
+            "C is the number of channels, and H and W are the "
+            "height and width. Note that this is for the 2D image. "
+            "Otherwise the size is (N x C x D1 x D2 ... x Dn)",
+            "T1")
+        .Input(
+            1,
+            "W",
+            "The weight tensor that will be used in the "
+            "convolutions; has size (M x C x kH x kW), where C "
+            "is the number of channels, and kH and kW are the "
+            "height and width of the kernel, and M is the number "
+            "of feature maps. For more than 2 dimensions, the "
+            "kernel shape will be (M x C x k1 x k2 x ... x kn), "
+            "where is the dimension of the kernel",
+            "T2")
+        .Input(
+            2,
+            "Z",
+            "padding value (zero_point normally), which should be a scalar tensor.",
+            "T1",
+            OpSchema::Optional)
+        .Output(
+            0,
+            "Y",
+            "Output data tensor that contains the result of the "
+            "convolution. The output dimensions are functions "
+            "of the kernel size, stride size, and pad lengths.",
+            "tensor(int32)")
+        .TypeConstraint(
+            "T1",
+            {"tensor(int8)",
+             "tensor(uint8)",
+             "tensor(int16)",
+             "tensor(uint16)"},
+            "Constrain input and output types to float tensors.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(int8)",
+             "tensor(uint8)",
+             "tensor(int16)",
+             "tensor(uint16)"},
+            "Constrain input and output types to float tensors.")
```

Comment:
Although allowing the weights to be distinct types from the input does allow flexibility, it also yields 16 (4x4) different combinations to handle :/.